### PR TITLE
Add Outscale CSI into rancher UI

### PR DESCRIPTION
**Pull request by outscale-mdr**
### Summary
This PR is to add [Outscale CSI](https://github.com/outscale/osc-bsu-csi-driver) into Rancher UI.

### Screenshot/Video

Preview:
Capture d’écran 2023-02-14 à 17 32 57
Proof:
Capture d’écran 2023-02-14 à 17 32 46

### CODE REVIEW

File: shell/edit/storage.k8s.io.storageclass/provisioners/bsu.csi.outscale.com.vue [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    keySize: {
+      get() {
+        return Number.parseInt(this.value.parameters['luks-key-size']);
+      },
+      set(value) {
+        this.$set(this.value.parameters, 'luks-key-size', value);
+      }
+    },
+    keySecrets: {
+      set(value) {
```
Comment (richard-cox) [inline shell/edit/storage.k8s.io.storageclass/provisioners/bsu.csi.outscale.com.vue:95]: There's no `get` here, so when it's used in the `v-model` below there's lots of errors (on load and use)

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: shell/models/storage.k8s.io.storageclass.js [hunk 2]
```diff
@@ -78,6 +78,11 @@ export const PROVISIONER_OPTIONS = [
     value:         'driver.harvesterhci.io',
     supported:     true,
     hideCustomize: true,
+  },
+  {
+    labelKey:  'storageClass.outscale-bsu.title',
+    value:     'bsu.csi.outscale.com',
+    supported: true,
```
Comment (richard-cox) [inline shell/models/storage.k8s.io.storageclass.js:85]: sorry, additional comment, supported will need to be removed as per https://www.suse.com/support/kb/doc/?id=000020481

Unsupported storage classes can be made visible by enabling the `unsupported-storage-drivers` Feature Flag (burger menu --> Global Setting --> Feautre Flags)

**Comment by outscale-mdr** [comment]
Hi @richard-cox,

Thank you for the feedback, I will make the changes as soon as possible !

**Comment by richard-cox** [comment]
@outscale-mdr Just an FYI, the cut off date for the next feature release is very roughly in a week

**Comment by outscale-mdr** [comment]
Hello @richard-cox,

Sorry for the silence, I was off a long time. Here are the changes. if it is not still possible to include it in the next release, the one after is fine.

Thank you !

**Review by richard-cox — CHANGES_REQUESTED** [review]
I hit some issues with the LabelSelect that used to selec the secret key. Below should be the fix, can you confirm?

```
- shell/edit/storage.k8s.io.storageclass/provisioners/bsu.csi.outscale.com.vue -
index 6e87bd136..dc1d06f86 100644
@@ -24,7 +24,10 @@ export default {
   async fetch() {
     const secrets = await this.$store.dispatch('cluster/findAll', { type: SECRET });
 
-    this.secrets = secrets.filter(s => s._type === TYPES.OPAQUE).map(s => s.id).sort();
+    this.secrets = secrets.filter(s => s._type === TYPES.OPAQUE).map(s => ({
+      label: s.id,
+      value: s.id
+    })).sort((a, b) => a.label.localeCompare(b.label));
   },
   data() {
     const canListStorageClasses = this.$store.getters['cluster/canList'](SECRET);
@@ -93,7 +96,10 @@ export default {
     },
     keySecrets: {
       get() {
-        return `${ this.value.parameters['csi.storage.k8s.io/node-stage-secret-namespace'] }/${ this.value.parameters['csi.storage.k8s.io/node-stage-secret-name'] }`;
+        const ns = this.value.parameters['csi.storage.k8s.io/node-stage-secret-namespace'];
+        const name = this.value.parameters['csi.storage.k8s.io/node-stage-secret-name'];
+
+        return ns && name ? `${ ns }/${ name }` : this.secrets[0];
       },
       set(value) {
         const secret = value.split('/', 2);
@@ -143,7 +149,7 @@ export default {
           :options="encryptionOptions"
         />
         <LabeledSelect
-          v-if="canListStorageClasses && value.parameters.encrypted === 'true'"
+          v-if="canListStorageClasses && value.parameters.encrypted === 'true' && secrets.length"
           v-model="keySecrets"
           :options="secrets"
           :label="t('storageClass.outscale-bsu.encryption.secret')"
```

I also couldn't create a secret, regardless of what was in the form.  PostS to `k8s/clusters/<CLUSTER>/v1/storage.k8s.io.storageclasses` always failed with  400 `StorageClass in version \"v1\" cannot be handled as a StorageClass: json: cannot unmarshal number into Go struct field StorageClass.parameters of type string`

**Comment by outscale-mdr** [comment]
Hello @richard-cox, 

Could we pause this PR to next release, I do not have time to dig into what is going on ?
In a few weeks, I will be able to have more time to focus on this PR.

Sorry about that

**Comment by richard-cox** [comment]
No problem, just let us know if you have any questions

**Comment by richard-cox** [comment]
@outscale-mdr Just an fyi, the timeframe for the next release is shorter than usual. There's about 3-4 weeks before we approach feature complete.

**Comment by richard-cox** [comment]
Closing due to inactivity. Please re-open when ready to update

**Pull request closed without merging**